### DISCUSSION TITLE
RFC - [DataGrid] Change how column options menu and filter are shown

### DISCUSSION THREAD

**Discussion by vnbaaij** [microsoft/fluentui-blazor#2288] [General]
## Current situation
With the DataGrid it is possible to use the `ColumnOptions` parameter to specify (usually) a filter for a column. In the current version, it looks like this (example comes from the demo site):

![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/ba54d01a-fdd5-4bbb-95ee-c8dced6134c6)

When a filter is applied, the indicator icon is changed to make clear this is a filtered column:

![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/a6957c05-c3ae-4dd8-a3ac-48d7bf41e51a)

The `ColumnOptions` can potentially be used for more things (see PR #2238) and showing a filter icon does not make a lot of sense then. 

## Proposed change
I propose to change this so that when `ColumnOptions` are specified, a generic 'more' icon will be shown. When `ColumnOptions` allows for a filter to be applied, a filter indicator will be shown besides the title akin to to `sort` indicator:

![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/7503b120-b706-4e07-bc14-7c9835e83e47)

When no filter is applied, no indicator will be shown of course (just as no sort indicator is shown when no sorting is taking place on a column)

![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/fe89dfad-34cc-4b75-880b-acba1ad6843e)

Thoughts? Questions? Remarks?

**Comment by juanmangd** [comment]
What option can be though for the user to notice that there is a sorting capability on the column header in question? Filter may be obvious given the input field but sorting may not be that obvious

**Reply by vnbaaij** [reply]
The whole header cell, besides the filter button (now) or more button (after change) is a button when a column can be sorted. That is already the case now and won't change.

**Comment by dvoituron** [comment]
I think that's great. Just a few questions:
1. Shouldn't we take the opportunity to adapt the "sort" icon, to make it softer, like FluentUI React does: 
![image](https://github.com/microsoft/fluentui-blazor/assets/8350694/644087d5-9f5a-45a2-a9b4-5f9023d526f0)
2. I may have confused the vertical [...] indicator with the Column Resize icon. But I don't know which one to use. A "ChevronDownUp" or other chevron? :-)

**Reply by vnbaaij** [reply]
> I think that's great. Just a few questions:
> 
> 1. Shouldn't we take the opportunity to adapt the "sort" icon, to make it softer, like FluentUI React does:

Yes, we can certainly do that

> 2. I may have confused the vertical [...] indicator with the Column Resize icon. But I don't know which one to use. A "ChevronDownUp" or other chevron? :-)

I'll try some variations.

**Reply by vnbaaij** [reply]
Something more like this:
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/d0db6558-f7e9-467a-adc1-d7938191a286)

**Reply by dvoituron** [reply]
Personally, I find it clearer :-)

**Reply by vnbaaij** [reply]
Me too!
